examples/iiqka_moveit_task_constructor_example/kuka_moveit_task_constructor/src/mtc_depalletizing_task_node.cpp

```diff
@@ -0,0 +1,314 @@
+        // Define Move Stage
+        auto move_stage = std::make_unique<mtc::stages::MoveTo>(
+          "move_" + std::to_string(4 * k + 2 * j + i), sampling_planner);
+        move_stage->setTimeout(10.0);
+        move_stage->properties().configureInitFrom(mtc::Stage::PARENT, {"group"});
+        move_stage->properties().set("marker_ns", "move_" + std::to_string(4 * k + 2 * j + i));
+        geometry_msgs::msg::PoseStamped move_pose;
+        move_pose.header.frame_id = "world";
+        move_pose.pose.position.x = -0.3 - PALLET_DISTANCE * i;
+        move_pose.pose.position.y = 0.1 - PALLET_DISTANCE * j;
+        move_pose.pose.position.z = 0.4;
+        move_pose.pose.orientation.x = 1.0;
+        move_pose.pose.orientation.y = 0.0;
+        move_pose.pose.orientation.z = 0.0;
+        move_pose.pose.orientation.w = 0.0;
+        move_stage->setGoal(move_pose);
+        task.add(std::move(move_stage));
+
+        // Define Place Stage
+        auto place_stage = std::make_unique<mtc::stages::MoveRelative>(
+          "place_" + std::to_string(4 * k + 2 * j + i), cartesian_planner);
+        place_stage->properties().set("marker_ns", "place_" + std::to_string(4 * k + 2 * j + i));
+        place_stage->properties().set("link", eef_frame);
+        place_stage->properties().configureInitFrom(mtc::Stage::PARENT, {"group"});
+        geometry_msgs::msg::Vector3Stamped vec_place;
+        vec_place.header.frame_id = "world";
+        vec_place.vector.z = -(2 - k) * PALLET_DISTANCE - 0.02;
+        place_stage->setDirection(vec_place);
+        task.add(std::move(place_stage));
+
+        // Detach Object Stage
+        auto detach_stage = std::make_unique<mtc::stages::ModifyPlanningScene>(
+          "detach object_" + std::to_string(4 * k + 2 * j + i));
+        detach_stage->detachObject("pallet_" + std::to_string(4 * k + 2 * j + i), eef_frame);
+        task.add(std::move(detach_stage));
+
+        // Define Lift2 Stage
+        auto lift2_stage = std::make_unique<mtc::stages::MoveRelative>(
+          "lift2_" + std::to_string(4 * k + 2 * j + i), cartesian_planner);
+        lift2_stage->properties().set("marker_ns", "lift2_" + std::to_string(4 * k + 2 * j + i));
+        lift2_stage->properties().set("link", eef_frame);
+        lift2_stage->properties().configureInitFrom(mtc::Stage::PARENT, {"group"});
+        geometry_msgs::msg::Vector3Stamped vec_lift2;
+        vec_lift2.header.frame_id = "world";
+        vec_lift2.vector.z = (2 - k) * PALLET_DISTANCE + 0.02;
+        lift2_stage->setDirection(vec_lift2);
+        task.add(std::move(lift2_stage));
+      }
+    }
+  }
+
+  return task;
+}
+
+bool MTCDepalletizingTaskNode::doTask(mtc::Task & task)
+{
+  if (task.stages()->numChildren() == 0)
+  {
+    RCLCPP_ERROR_STREAM(LOGGER, "No task set.");
+    return false;
+  }
+
+  try
+  {
+    task.init();
+  }
+  catch (mtc::InitStageException & e)
+  {
+    RCLCPP_ERROR_STREAM(LOGGER, e);
+    return false;
+  }
+
+  if (!task.plan(1000))
```

Comment:
just out of curiosity, does this plan all steps at the same time? if yes, how is it possible that the robot does not collide with the upper pallets when going for the lower ones?

Reply:
It creates them one after the other. The console / rviz plugin shows the number of successful and failed solution for every step. It would be cool if it could plan all steps at the same time tho. :) (Too bad it would not know the exact start pose for some paths)

Reply:
Also, MTC is well integrated with the planning scene.

Reply:
aaa, okay, this perfectly makes sense, (I just somehow missed the while loop)